src/main/java/reposense/parser/CsvParser.java

```diff
@@ -0,0 +1,126 @@
+            }
+
+        } catch (IOException ioe) {
+            throw new IOException(MESSAGE_UNABLE_TO_READ_CSV_FILE);
+        }
+
+        return repoConfigurations;
+    }
+
+    /**
+     * Extracts {@code Author} information from the {@code line}.
+     */
+    private static void processLine(List<RepoConfiguration> repoConfigurations,
+            Date sinceDate, Date untilDate, String line, int lineNumber) {
+        if (!line.isEmpty()) {
+            String[] elements = line.split(Constants.CSV_SPLITTER);
+
+            if (elements.length < GITHUB_ID_POSITION) {
+                // Warns malformed line and skips it
+                System.out.println(String.format(MESSAGE_MALFORMED_LINE_FORMAT, lineNumber, line));
+                return;
+            }
+
+            String organization = elements[ORGANIZATION_POSITION];
+            String repositoryName = elements[REPOSITORY_NAME_POSITION];
+            String branch = elements[BRANCH_POSITION];
+
+            RepoConfiguration config = new RepoConfiguration(organization, repositoryName, branch);
+            int index = repoConfigurations.indexOf(config);
+
+            if (index != -1) {
+                config = repoConfigurations.get(index);
+            } else {
+                repoConfigurations.add(config);
+                config.setSinceDate(sinceDate);
+                config.setUntilDate(untilDate);
+            }
+
+            Author author = new Author(elements[GITHUB_ID_POSITION]);
+            config.getAuthorList().add(author);
+            setDisplayName(elements, config, author);
+            setAlias(elements, config, author);
+        }
+    }
+
+    private static void setDisplayName(String[] elements, RepoConfiguration config, Author author) {
+        // Checks length of the elements array as trailing commas may be omitted for empty columns
+        if (elements.length > DISPLAY_NAME_POSITION && !elements[DISPLAY_NAME_POSITION].isEmpty()) {
+            //Not empty, take the supplied value as Display Name
+            config.getAuthorDisplayNameMap().put(author, elements[DISPLAY_NAME_POSITION]);
+        } else {
+            //else, use GitHub Id as Display Name
+            config.getAuthorDisplayNameMap().put(author, author.getGitId());
+        }
+    }
+
+    private static void setAlias(String[] elements, RepoConfiguration config, Author author) {
+        //Always use GitHub Id as an alias
+        config.getAuthorAliasMap().put(elements[GITHUB_ID_POSITION], author);
+
+        // Checks length of the elements array as trailing commas may be omitted for empty columns
+        // If more alias are provided, use them as well
+        if (elements.length > ALIAS_POSITION &&  !elements[ALIAS_POSITION].isEmpty()) {
+            String[] aliases = elements[ALIAS_POSITION].split(Constants.AUTHOR_ALIAS_SPLITTER);
+
+            for (String alias : aliases) {
+                config.getAuthorAliasMap().put(alias, author);
+            }
+        }
```

Comment:
In general, try to make the code self-explanatory (e.g., [use explaining variables](https://refactoring.com/catalog/extractVariable.html)) to reduce the need for explanatory comments.

Reply:
Prof, I made the changes. Please kindly review.